dtfs-central-api/src/v1/controllers/utilisation-report-service/post-keying-data.controller/helpers.test.ts

```diff
@@ -0,0 +1,116 @@
+import { when } from 'jest-when';
+import { FeeRecordEntityMockBuilder, UtilisationReportEntityMockBuilder } from '@ukef/dtfs2-common';
+import { FeeRecordRepo } from '../../../../repositories/fee-record-repo';
+import { getGenerateKeyingDataDetails } from './helpers';
+
+describe('post-keying-data.controller helpers', () => {
+  describe('getGenerateKeyingDataDetails', () => {
+    const findByStatusesSpy = jest.spyOn(FeeRecordRepo, 'findByStatuses');
+
+    const RECONCILIATION_IN_PROGRESS_REPORT = UtilisationReportEntityMockBuilder.forStatus('RECONCILIATION_IN_PROGRESS').build();
+
+    const aMatchFeeRecord = (facilityId: string) =>
+      FeeRecordEntityMockBuilder.forReport(RECONCILIATION_IN_PROGRESS_REPORT).withStatus('MATCH').withFacilityId(facilityId).build();
+
+    const aToDoFeeRecord = (facilityId: string) =>
+      FeeRecordEntityMockBuilder.forReport(RECONCILIATION_IN_PROGRESS_REPORT).withStatus('TO_DO').withFacilityId(facilityId).build();
+
+    beforeEach(() => {
+      findByStatusesSpy.mockRejectedValue(new Error('Some error'));
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it('returns each supplied fee record with generateKeyingData set to true when there are no facility ids which are not at the TO_DO or DOES_NOT_MATCH status', async () => {
```

Comment:
Too many negatives, should be: `when there are no facility ids which are at the TO_DO or DOES_NOT_MATCH status`